ext/standard/file.c

```diff
@@ -913,8 +913,8 @@ PHPAPI PHP_FUNCTION(fclose)
 	PHP_STREAM_TO_ZVAL(stream, res);
 
 	if ((stream->flags & PHP_STREAM_FLAG_NO_FCLOSE) != 0) {
-		php_error_docref(NULL, E_WARNING, "%d is not a valid stream resource", stream->res->handle);
-		RETURN_FALSE;
+		zend_type_error("%d is not a valid stream resource", stream->res->handle);
```

Comment:
Not sure whether this should be converted to an exception.

Reply:
It seems to me that we already do those kind of type errors, can you confirm @nikic ?

Reply:
Well, throwing on a completely unrelated resource (such as a GD font resource) makes sense, but I'm not sure about closing an uncloseable stream.

Reply:
Me neither, so, I'll probably revert that specific change and commit in a week if there are no other objections.

Reply:
Yeah, it's probably better to leave this alone, especially as I don't think there's currently even a way to determine whether this flag is set. Recent example where it is used: https://github.com/php/php-src/commit/e36daa6927c05d2e687bb77495ef206cde118b33

Reply:
ACK